plugin-server/src/cdp/cdp-api.ts

```diff
@@ -388,4 +393,76 @@ export class CdpApi {
                 error: 'Method not allowed',
             })
         }
+
+    private postMailjetWebhook =
+        () =>
+        async (req: express.Request & { rawBody?: Buffer }, res: express.Response): Promise<any> => {
+            const signature = req.headers['x-mailjet-signature'] as string
+            const timestamp = req.headers['x-mailjet-timestamp'] as string
+
+            if (!signature || !timestamp || !req.rawBody) {
+                return res.status(403).send('Missing required headers or body')
+            }
+
+            const payload = `${timestamp}.${req.rawBody.toString()}`
+            const hmac = crypto.createHmac('sha256', this.MAILJET_SECRET).update(payload).digest()
+
+            try {
+                const signatureBuffer = Buffer.from(signature, 'hex')
+                if (
+                    hmac.length !== signatureBuffer.length ||
+                    !crypto.timingSafeEqual(new Uint8Array(hmac), new Uint8Array(signatureBuffer))
+                ) {
+                    return res.status(403).send('Invalid signature')
+                }
+
+                // Track Mailjet webhook metrics
+                const event = req.body as MailjetEvent
+                const category = eventTypeToCategory[
+                    event.event as keyof typeof eventTypeToCategory
+                ] as AppMetric['category']
+
+                if (event) {
+                    const metric: AppMetric = {
```

Comment:
Confusing but we have AppMetric and AppMetric2 - we should only be using the new one (if you ever import code from `worker` then you should stop 😅 that stuff is all super legacy.